hredis/redigo/redigo.go

```diff
@@ -42,16 +45,129 @@ func WaitForAvailability(url string, d time.Duration, f WaitFunc) (bool, error)
 	}
 }
 
+// DialURLFunc describes the type of redis.DialURL, useful for changing the behavior
+// of a redis Dialer.
+type DialURLFunc func(string, ...redis.DialOption) (redis.Conn, error)
+
+// RedisDialer is an internal helper providing custom Redis dial logic for
```

Comment:
If it's internal, why is it exported?

Reply:
I was thinking that in order to use `WithPasswords` you'd need the type exported, as part of the type is a function that exposes it. But, I'm likely wrong here.

Reply:
Yup. I'm wrong. :)